gcc/rust/typecheck/rust-tyty.cc

```diff
@@ -106,6 +106,29 @@ InferType::clone ()
 			get_combined_refs ());
 }
 
+bool
+InferType::default_type (BaseType **type) const
+{
+  auto context = Resolver::TypeCheckContext::get ();
+  bool ok = false;
+  switch (infer_kind)
+    {
+    case GENERAL:
+      break;
+      case INTEGRAL: {
+	ok = context->lookup_builtin ("i32", type);
```

Comment:
this could be simpler with just return context->lookup and remove the assertions

Reply:
Actually, keep the assertions but you can return ok instead of break. Since this is very unlikely to fail but incase it does an assertion will help alot.